scripts/pi-hole/php/customdns.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+    require_once "func.php";
+
+    $customDNSFile = "/etc/pihole/custom.list";
+
+    switch ($_REQUEST['action'])
+    {
+        case 'get':     echo json_encode(echoCustomDNSEntries());    break;
+        case 'add':     echo json_encode(addCustomDNSEntry());      break;
+        case 'delete':  echo json_encode(deleteCustomDNSEntry());   break;
+        default:
+            die("Wrong action");
+    }
+
+    function echoCustomDNSEntries()
+    {
+        $entries = getCustomDNSEntries();
+
+        $data = [];
+
+        foreach ($entries as $domain => $ip)
+            $data[] = [ $domain, $ip ];
+
+        return [ "data" => $data ];
+    }
+
+    function getCustomDNSEntries()
+    {
+        global $customDNSFile;
+
+        $entries = [];
+
+        $handle = fopen($customDNSFile, "r");
+        if ($handle)
+        {
+            while (($line = fgets($handle)) !== false) {
+                $line = str_replace("\r","", $line);
+                $line = str_replace("\n","", $line);
+                $explodedLine = explode (" ", $line);
+
+                if (count($explodedLine) != 2)
+                    continue;
+
+                $entries[$explodedLine[1]] = $explodedLine[0];
+            }
+
+            fclose($handle);
+        }
+
+        return $entries;
+    }
+
+    function addCustomDNSEntry()
+    {
+        try
+        {
+            $ip = !empty($_REQUEST['ip']) ? $_REQUEST['ip']: "";
+            $domain = !empty($_REQUEST['domain']) ? $_REQUEST['domain']: "";
+
+            if (empty($ip))
+                return errorJsonResponse("IP must be set");
+
+            if (!filter_var($ip, FILTER_VALIDATE_IP))
+                return errorJsonResponse("IP must be valid");
+
+            if (empty($domain))
+                return errorJsonResponse("Domain must be set");
+
+            if (!is_valid_domain_name($domain))
+                return errorJsonResponse("Domain must be valid");
+
+            $existingEntries = getCustomDNSEntries();
+
+            if (array_key_exists($domain, $existingEntries))
```

Comment:
This should take care of the case someone wants to add multiple IP addresses for the same domain, e.g. separate IPv4 and IPv6 address.